lib/selector.php

```diff
@@ -226,7 +226,7 @@ private static function getPages()
         if (rex_addon::get('structure')->isAvailable()) {
 
             $sql = rex_sql::factory();
-            $pages = $sql->getArray('SELECT id,clang_id FROM ' . rex::getTablePrefix() . 'article WHERE status = 1', array(), PDO::FETCH_NUM);
+            $pages = $sql->getArray('SELECT a.id, a.clang_id FROM ' . rex::getTable('article') . ' AS a LEFT JOIN ' . rex::getTable('clang') . ' AS c ON a.clang_id = c.id WHERE a.status = ? AND c.status = ?', [1, 1], PDO::FETCH_NUM);
```

Comment:
INNER JOIN liefert nur datensätze die in beiden tabellen vorhanden sind. das ist hier für alle datensätze der falle.

OUTER JOIN nimmt man wenn man 2 tabellen verbindet, es aber nicht zu jedem Datensatz in der FROM tabelle auch einen Datensatz in der zu join'nenden Tabelle gibt (wäre in unserem Fall: Falls es Artikel gäbe die keine Sprache zugeordnet wären).
d.h. OUTER JOIN liefert auch die die keine zuordnung haben. INNER JOIN liefert nur welche die ein Pendant in beiden Tabellen haben.